tests/integration/buildcmd/test_build_terraform_applications.py

```diff
@@ -84,3 +128,137 @@ def test_exit_success_no_beta_features_flags_supplied_hooks(self):
+        )
+        LOG.info("command list: %s", build_cmd_list)
+        _, stderr, return_code = self.run_command(build_cmd_list)
+        LOG.info(stderr)
+        self.assertEqual(return_code, 0)
+
+        self._verify_invoke_built_function(
+            template_path=str(Path(self.terraform_application_path) / ".aws-sam/build/template.yaml"),
+            function_logical_id=function_identifier,
+            overrides=None,
+            expected_result={"statusCode": 200, "body": "[]"},
+        )
+
+
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+class TestBuildTerraformApplicationsWithZipBasedLambdaFunctionAndS3Backend(BuildTerraformApplicationIntegBase):
+    terraform_application = Path("terraform/zip_based_lambda_functions_s3_backend")
+    functions = [
+        "aws_lambda_function.from_localfile",
+        "aws_lambda_function.from_s3",
+        "module.level1_lambda.aws_lambda_function.this",
+        "module.level1_lambda.module.level2_lambda.aws_lambda_function.this",
+        "my_function_from_localfile",
+        "my_function_from_s3",
+        "my_level1_lambda",
+        "my_level2_lambda",
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        """Setting up a S3 bucket (using pre-created or create a new bucket) to use as a Terraform backend"""
+        cls.region_name = os.environ.get("AWS_DEFAULT_REGION", "us-east-1")
+
+        """Please read comments in package_integ_base.py for more details around this."""
+        bucket_env_var = os.environ.get("AWS_S3")
+        cls.pre_created_bucket = False
+        if bucket_env_var:
+            cls.pre_created_bucket = os.environ.get(bucket_env_var, False)
+        cls.bucket_name = cls.pre_created_bucket if cls.pre_created_bucket else str(uuid.uuid4())
+
+        s3 = boto3.resource("s3")
+        cls.s3_bucket = s3.Bucket(cls.bucket_name)
+        if not cls.pre_created_bucket:
+            cls.s3_bucket.create()
+            time.sleep(S3_SLEEP)
+
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean up and delete the bucket if it is not pre-created"""
+        cls.s3_bucket.objects.all().delete()
+        time.sleep(S3_SLEEP)
+        if not cls.pre_created_bucket:
+            cls.s3_bucket.delete()
+
+    def setUp(self):
+        self.backend_key = str(Path("terraform-backend") / str(uuid.uuid4()))
+        self.backendconfig_path = str(Path(self.terraform_application_path) / "backend.conf")
+        with open(self.backendconfig_path, "w") as f:
+            f.write(f'bucket="{self.bucket_name}"\n')
+            f.write(f'key="{self.backend_key}"\n')
+            f.write(f'region="{self.region_name}"')
+
+        # We have to init the terraform project with specifying the S3 backend first
+        _, stderr, _ = self.run_command(
+            ["terraform", "init", f"-backend-config={self.backendconfig_path}", "-reconfigure"]
```

Comment:
Is `terraform` already configured in Appveyor?

Reply:
yes